test/connection.js

```diff
@@ -0,0 +1,51 @@
+/* eslint-env node, mocha */
+require('./init.js');
+var assert = require('assert');
+var DataSource = require('loopback-datasource-juggler').DataSource;
+
+var config;
+
+before(function() {
+  config = global.getConfig();
+  console.log("config", config);
+});
+
+describe('testConnection', function() {
+  it('should pass with valid settings', function(done) {
+    var db = new DataSource(require('../'), config);
+    db.ping(function(err) {
+      assert(!err, 'Should connect without err.');
```

Comment:
I see another done(err) on the line under. Maybe I'm confused.

But if it looks like this:

```
if (err) done(err);
done();
```

You'll call done twice which is not what you want.